### DISCUSSION TITLE
VPN Check

### DISCUSSION THREAD

**Discussion by MarkMain** [transmission/transmission#4579] [Ideas]
Can you add a settings option that requires a VPN connection before proceeding with transmission activity?
Can you also halt activity if the VPN connection is accidentally turned off?
Are you able to automatically run the VPN with a default connection?

Since a VPN is so important, it would be a nice option to require its use based on my settings preference.

Thanks for considering the idea.

**Comment by josephj11** [comment]
This would be nice to have.

As a workaround (on Linux), I wrote a bash wrapper script for Transmission that refuses to start it if my VPN is not active. I wrote another bash script that runs as a daemon in the background and checks the VPN status every five seconds and kills Transmission if the VPN is not active. When it does so, it keeps running and restarts Transmission when the VPN becomes active again.

Scripts available (as is) upon request.

**Comment by andrewleber** [comment]
This can be configured in most VPN clients as a _kill switch_ that automatically disables your connectivity if the VPN connection fails. Some providers, like NordVPN, provide granular application-level kill switches so you can only enable it for Transmission, while others are system-level.

**Reply by josephj11** [reply]
Unfortunately, my VPN doesn't have that feature.

**Reply by andrewleber** [reply]
You can try using the steps described here to jerry-rig one yourself: https://www.privacyaffairs.com/vpn-killswitch/

Good luck!

**Reply by andrewleber** [reply]
Oh, just re-read your original reply and saw that you already jerry-rigged one. Nice job

**Comment by wally-crunk** [comment]
qBittorrent has this feature, and it would be nice to have on Transmission as a built in setting.
